pkg/insights/insightsreport/insightsreport.go

```diff
@@ -0,0 +1,184 @@
+package insightsreport
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog"
+
+	"github.com/openshift/insights-operator/pkg/authorizer"
+	"github.com/openshift/insights-operator/pkg/config"
+	"github.com/openshift/insights-operator/pkg/controllerstatus"
+	"github.com/openshift/insights-operator/pkg/insights/insightsclient"
+)
+
+// Gatherer gathers the report from Smart Proxy
+type Gatherer struct {
+	controllerstatus.Simple
+
+	configurator          Configurator
+	client                *insightsclient.Client
+	LastReport            SmartProxyReport
+	archiveUploadReporter <-chan struct{}
+	retries               int64
+}
+
+// Response represents the Smart Proxy report response structure
+type Response struct {
+	Report SmartProxyReport `json:"report"`
+}
+
+// Configurator represents the interface to retrieve the configuration for the gatherer
+type Configurator interface {
+	Config() *config.Controller
+	ConfigChanged() (<-chan struct{}, func())
+}
+
+// InsightsReporter represents an object that can notify about archive uploading
+type InsightsReporter interface {
+	ArchiveUploaded() <-chan struct{}
+}
+
+var (
+	// ErrWaitingForVersion contains an error when the cluster version is not loaded yet
+	ErrWaitingForVersion = fmt.Errorf("waiting for the cluster version to be loaded")
+
+	// insightsStatus contains a metric with the latest report information
+	insightsStatus = metrics.NewGaugeVec(&metrics.GaugeOpts{
+		Namespace: "health",
+		Subsystem: "statuses",
+		Name:      "insights",
+		Help:      "Information about the cluster health status as detected by Insights tooling.",
+	}, []string{"metric"})
+)
+
+// New initializes and returns a Gatherer
+func New(client *insightsclient.Client, configurator Configurator, reporter InsightsReporter) *Gatherer {
+	return &Gatherer{
+		Simple:                controllerstatus.Simple{Name: "insightsreport"},
+		configurator:          configurator,
+		client:                client,
+		archiveUploadReporter: reporter.ArchiveUploaded(),
+		retries:               0,
+	}
+}
+
+// PullSmartProxy performs a request to the Smart Proxy and unmarshal the response
+func (r *Gatherer) PullSmartProxy() (bool, error) {
+	klog.Info("Pulling report from smart-proxy")
+	config := r.configurator.Config()
+	initialWait := config.ReportPullingDelay
+	minimalRetryTime := config.ReportMinRetryTime
+	reportEndpoint := config.ReportEndpoint
+
+	if len(reportEndpoint) == 0 {
+		klog.V(4).Info("Not downloading report because Smart Proxy client is not properly configured: missing report endpoint")
+		return true, nil
+	}
+
+	delay := initialWait - time.Duration(r.retries*minimalRetryTime.Nanoseconds()) - minimalRetryTime
```

Comment:
Is it intentionally waiting 60s - (0 x 10s) - 10s, 60 - (1 x 10s) - 10s, 60 - (2x10s) - 10s etc ?
what about 
```
pause := wait(retry int) time.Duration { 
  if retry ==0 {
  return wait.Jitter(initialWait, 0.1)
}
  return wait.Jitter(minRetryTime, 0.3)
}
 if wait > 0 {
  time.Sleep(pause)
}
```
To wait for 60 + rand(6s), 10 + rand(3s), 10 + rand(3)

Reply:
I will use Jitter in next commit. TYVM